### PR TITLE
Issue #11298: Added switch statements and lambda support to WhitespaceAfter

### LINKED ISSUE #11298: WhitespaceAfter does not work on 'switch' and lambda statements
I want to be able to configure Checkstyle to report a violation on missing space after 'switch', which is currently not possible.
WhitespaceAfter detects a missing space after 'for', 'if' etc. but _not_ after 'switch'.
It would be nice if there would be a LITERAL_SWITCH similar to LITERAL_FOR etc.

https://checkstyle.org/config_whitespace.html#WhitespaceAfter
I have downloaded the latest cli from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

**How it works Now:**

```bash
peter@pangolin:~/Downloads/checkstyle$ javac Test.java 

peter@pangolin:~/Downloads/checkstyle$ cat config.xml 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
	"-//Puppy Crawl//DTD Check Configuration 1.3//EN"
	"http://checkstyle.sourceforge.net/dtds/configuration_1_3.dtd">
<module name="Checker">
	<module name="TreeWalker">
		<module name="WhitespaceAfter"/>
	</module>
</module>

peter@pangolin:~/Downloads/checkstyle$ cat Test.java 
class Test {

	void test() {
		for(int i = 0; i < 10; i++) { // Checkstyle catches this, configurable with LITERAL_FOR.
		}
		switch(1) { // Checkstyle does NOT catch this, there is no LITERAL_SWITCH to configure.
			default:
				break;
		}
	}
}

peter@pangolin:~/Downloads/checkstyle$ RUN_LOCALE="-Duser.language=en -Duser.country=US"
peter@pangolin:~/Downloads/checkstyle$ java $RUN_LOCALE -jar checkstyle-9.3-all.jar -c config.xml Test.java
Starting audit...
[ERROR] /home/peter/Downloads/checkstyle/Test.java:4:17: 'for' is not followed by whitespace. [WhitespaceAfter]
Audit done.
Checkstyle ends with 1 errors.
```

The problem is that WhitespaceAfter catches missing spaces after 'if', 'for' etc., but _not_ after 'switch'.
Similar to 'for', I suggest a configurable token LITERAL_SWITCH with the "token" property which is configured by default, and a similar output:

```bash
[ERROR] home/peter/Downloads/checkstyle/Test.java:6:21: 'switch' is not followed by whitespace. [WhitespaceAfter]
```

**Additional context**

The code style that I want to impose:

```java
for (int i = 0; i < 10; i++) {
}
switch (1) {
	default:
		break;
}
```

### PR DESCRIPTION
**Pull request by Vyom-Yadav**
Resolves #11298 
Diff Regression config: https://gist.githubusercontent.com/Vyom-Yadav/98dceb63a79f4833e85fff9b2e1464a6/raw/97410d450226ecfe8c45df4f2693cd61c25c3765/my_checks.xml

Diff Regression patch config: https://gist.githubusercontent.com/Vyom-Yadav/273327b9c6cd6963b7dade792bc2c9c2/raw/a2715eb5a51a57dec2e34472a9797abf85f267db/patch_config.xml

---
PMD PR - https://github.com/pmd/pmd/pull/3767
Orekit PR - https://github.com/CS-SI/Orekit/pull/29

### CODE REVIEW

**Comment by Vyom-Yadav** [comment]
I didn't create a new issue for lambda expressions as enhanced switch also uses the `LAMBDA` token.

**Comment by Vyom-Yadav** [comment]
Github, generate report

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java [hunk 1]
```diff
@@ -153,6 +157,8 @@
             TokenTypes.LITERAL_FOR,
             TokenTypes.DO_WHILE,
             TokenTypes.ELLIPSIS,
+            TokenTypes.LITERAL_SWITCH,
```
Comment (strkkk) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java:160]: I can agree on lambda to be included in list of default tokens, but switch is not that obvious. I think we can exclude it from list of default tokens. Report shows that it is pretty common to do not put whitespace after switch.

@romani @rnveach @nmancus1 @pbludov what do you think?

Comment (Vyom-Yadav) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java:160]: WhitespaceAround also contains `LITERAL_SWITCH`, we can put it in acceptable tokens if not in default.
https://github.com/checkstyle/checkstyle/blob/4b568976ada9210363162e64e232a0c67edbdc74/src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAroundCheck.java#L527

Comment (pbludov) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java:160]: > Report shows that it is pretty common to do not put whitespace after switch.

I don't think it's intentional. For example, HBase uses `switch (` too: https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/916c493_2022100915/reports/diff/Hbase/xref/home/runner/work/checkstyle/checkstyle/contribution/checkstyle-tester/repositories/Hbase/hbase-examples/src/main/java/org/apache/hadoop/hbase/coprocessor/example/generated/BulkDeleteProtos.java.html#L98

The `switch` is the same statement as `for`, `while`, `catch`. I see no reason to make an exception for it.

**Review by peterdemaeyer — APPROVED** [review]
Looks good to me.
I like the fact that you also covered the lambda.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java (hunk 1 above)
Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java:160]: https://google.github.io/styleguide/javaguide.html#s4.6.2-horizontal-whitespace demands space in similar structures, so I think it is ok to make is as default.

@Vyom-Yadav , please update inputs for Google style to make it clear our config will start to enforce space after such tokens.

Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java:160]: @pbludov , @rnveach please share your position on default

Comment (Vyom-Yadav) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java:160]: > please update inputs for Google style to make it clear our config will start to enforce space after such tokens.

Done

Comment (pbludov) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java:160]: > @pbludov , @rnveach please share your position on default

SWITCH should be included by default.

**Review by romani — APPROVED** [review]

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java (hunk 1 above)
Comment (strkkk) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java:160]: ok, thanks

**Review by strkkk — APPROVED** [review]

**Comment by romani** [comment]
Wercker detected regression on PMD, we need to send PR to them.
```
[INFO] --- maven-checkstyle-plugin:3.1.2:check (checkstyle-check) @ pmd-core ---
[INFO] There is 1 error reported by Checkstyle 10.0-SNAPSHOT with /net/sourceforge/pmd/pmd-checkstyle-config.xml ruleset.
[ERROR] src/main/java/net/sourceforge/pmd/cpd/GUI.java:[173,21] (whitespace) WhitespaceAfter: 'switch' is not followed by whitespace.
```

**Review by pbludov — APPROVED** [review]

**Comment by Vyom-Yadav** [comment]
@romani All related PR's have been merged

**Pull request merged**